src/TeamSyncer.ts

```diff
@@ -0,0 +1,317 @@
+                await new Promise((r) => setTimeout(
+                    r,
+                    ms,
+                ));
+            }
+        }
+        const queue = new PQueue({concurrency: TEAM_SYNC_CHANNEL_CONCURRENCY});
+        log.info(`Found ${channelList.length} total channels`);
+        return;
+        for (const channel of channelList) {
+            // tslint:disable-next-line: no-floating-promises
+            queue.add(() => this.syncChannel(teamId, channel));
+        }
+        await queue.onIdle();
+    }
+
+    public async syncUsers(teamId: string, client: WebClient) {
+        if (!this.shouldTeamSync(teamId, "user")) {
+            return;
+        }
+    }
+
+    private async shouldTeamSync(teamId: string, item?: "channel"|"user", itemId?: string) {
+        if (!this.teamConfigs[teamId]) {
+            return false;
+        }
+        if (this.teamConfigs[teamId].enabled) {
+            return false;
+        }
+        const channels = this.teamConfigs[teamId].channels;
+        if (item === "channel" && channels && itemId) {
+            if (channels.blacklist) {
+                if (channels.blacklist.includes(itemId)) {
+                    log.warn("Not bridging channel, blacklisted");
+                    return false;
+                }
+            }
+            if (channels.whitelist) {
+                if (!channels.whitelist.includes(itemId)) {
+                    log.warn("Not bridging channel, not in whitelist");
+                    return false;
+                }
+            }
+        }
+        // More granular please.
+        return true;
+    }
+
+    public async onChannelAdded(teamId: string, channelId: string, name: string, creator: string) {
+        // Should create the channel?
+        log.info(`${teamId} created channel ${channelId}`);
+        const existingChannel = this.main.rooms.getBySlackChannelId(channelId);
+        if (existingChannel) {
+            log.info("Channel already exists in datastore, not bridging");
+            return;
+        }
+        const client = await this.main.clientFactory.getTeamClient(teamId);
+        const { channel } = (await client.conversations.info({ channel: channelId })) as ConversationsInfoResponse;
+        if (!channel.is_channel || channel.is_private) {
+            log.warn("Not creating channel: Is either private or not a channel");
+            return;
+        }
+        const userId = (await this.main.datastore.getTeam(teamId))!.user_id;
+        const {user} = (await client.users.info({ user: userId })) as UsersInfoResponse;
+        await client.chat.postEphemeral({
+            user: creator,
+            text: `Hint: To bridge to Matrix, run the \`/invite @${user!.name}\` command in this channel.`,
+            channel: channelId,
+        });
+
+        // Create the room first.
+        try {
+            const roomId = await this.createRoomForChannel(teamId, creator, channel);
```

Comment:
can we also have a list of matrix users to invite to the new room in the config?